src/Data/Serialize/Put.hs

```diff
@@ -97,6 +97,9 @@ import qualified Data.ByteString.Short as BS
 
 import qualified Control.Applicative as A
 import Data.Array.Unboxed
+#if MIN_VERSION_base(4,10,0)
```

Comment:
FWIW, `semigroup` has been around since `base-4.9`, so we could backport these instances to that version.

Alternatively, we could depend on `semigroups` to have it defined even further back, but I suppose that's up to the maintainers' discretion.

Reply:
Fair point. I've bumped the bound down.